app/components/post_list/more_messages/more_messages.tsx

```diff
@@ -110,24 +118,44 @@ const MoreMessages = ({
     unreadCount,
     testID,
     theme,
-    currentCallBarVisible,
-    joinCallBannerVisible,
 }: Props) => {
     const serverUrl = useServerUrl();
-    const isTablet = useIsTablet();
     const insets = useSafeAreaInsets();
     const pressed = useRef(false);
     const resetting = useRef(false);
     const initialScroll = useRef(false);
     const [loading, setLoading] = useState(EphemeralStore.isLoadingMessagesForChannel(serverUrl, channelId));
     const [remaining, setRemaining] = useState(0);
     const underlayColor = useMemo(() => `hsl(${hexToHue(theme.buttonBg)}, 50%, 38%)`, [theme]);
+    const styles = getStyleSheet(theme);
     const top = useSharedValue(0);
-    const adjustedShownTop = SHOWN_TOP + (currentCallBarVisible ? CURRENT_CALL_BAR_HEIGHT : 0) + (joinCallBannerVisible ? JOIN_CALL_BAR_HEIGHT : 0);
-    const adjustTop = isTablet || (isCRTEnabled && rootId);
-    const shownTop = adjustTop ? SHOWN_TOP : adjustedShownTop;
+
+    // Calls state
+    const incomingCalls = useIncomingCalls().incomingCalls;
+    const channelsWithCalls = useChannelsWithCalls(serverUrl);
+    const callsState = useCallsState(serverUrl);
+    const globalCallsState = useGlobalCallsState();
+    const currentCall = useCurrentCall();
+    const dismissed = Boolean(callsState.calls[channelId]?.dismissed[callsState.myUserId]);
+    const inCurrentCall = currentCall?.id === channelId;
+    const joinCallBannerVisible = Boolean(channelsWithCalls[channelId]) && !dismissed && !inCurrentCall;
+
+    // Do we have calls banners?
+    const currentCallBarVisible = Boolean(currentCall);
+    const micPermissionsError = !globalCallsState.micPermissionsGranted && (currentCall && !currentCall.micPermissionsErrorDismissed);
+    const callQualityAlert = Boolean(currentCall?.callQualityAlert);
+    const incomingCallsShowing = incomingCalls.filter((ic) => ic.channelID !== channelId);
+    const callsIncomingAdjustment = (incomingCallsShowing.length * CALL_NOTIFICATION_BAR_HEIGHT) + (incomingCallsShowing.length * 8);
+    const callsAdjustment = (currentCallBarVisible ? CURRENT_CALL_BAR_HEIGHT + 8 : 0) +
+        (micPermissionsError ? CALL_ERROR_BAR_HEIGHT + 8 : 0) +
+        (callQualityAlert ? CALL_ERROR_BAR_HEIGHT + 8 : 0) +
+        (joinCallBannerVisible ? JOIN_CALL_BAR_HEIGHT + 8 : 0) +
+        callsIncomingAdjustment;
+
+    // The final top:
+    const adjustedTop = insets.top + callsAdjustment;
```

Comment:
Is there a chance this logic could live under `products/calls` and only plug a single function call on this side? Mostly concerned with all the places where Calls is expanding on Channels without a clear boundary.

Reply:
Yep, good idea, done in https://github.com/mattermost/mattermost-mobile/pull/7620/commits/4dd9b117d5699dc41f282a8bafc2588743bd17e2